### ISSUE TITLE
Components page: consider pulling examples out of iframes (maybe namespace classes another way?)

### ISSUE THREAD

**Issue by ericlindley-g** [ampproject/ampstart#282]
Current issues:
• Loading time is long for examples (see loading dots on first load)
• Component labels don't inherit styles of page

Possible solutions:
Pull component labels outside of iframe
Pull entire example outside of iframe, and encapsulate/namespace CSS some other way (shadow dom?)

**Comment by camelburrito** [comment]
No we cannot , because our website also uses ampstart. Namespacing would not solve this issue because baseline templates rely on the ampstart namespace. 

We could move the labels outside and make them left aligned, we cannot guarantee that they would be aligned wrt the rendered item (Hence we have it inside). We could however make it match the page styles easily. 

With browser caches turned on (which is normal) the iframes should load faster. 
Also there are too many assets that come with the component renderings (Images, some times required extensions). It is good that they are lazy loaded without affecting the performance of the components page. Also runtime does some memory management by pausing and un-pausing iframes which is good for us.

We will need to focus on getting https://github.com/ampproject/amphtml/issues/7570 fixed. so that iframes can resize properly and we can avoid the huge gaps.

**Comment by ericlindley-g** [comment]
Makes sense—I'll change this to just a request for the labels to use the parent page style
